PseudoGantry/lib/PseudoGantry/PseudoAxis.cpp

```diff
@@ -0,0 +1,121 @@
+#include "PseudoAxis.h"
+
+void set_up_encoder(PseudoEncoder *encoder, void (*isr_motor_pulse)(void))
+{
+    pinMode(encoder->motor_pulse_pin, INPUT_PULLUP);
+    pinMode(encoder->channel_a_out, OUTPUT);
+    digitalWrite(encoder->channel_a_out, LOW);
+    attachInterrupt(digitalPinToInterrupt(encoder->motor_pulse_pin), isr_motor_pulse, CHANGE);
+}
+
+bool toggle_encoder_output(PseudoAxis *pseudo_axis)
+{
+    pseudo_axis->skip_counter++;
+    // skip steps in the beginning
+    if (pseudo_axis->skip_counter <= pseudo_axis->changes_to_skip) return false;
+    // else start toggling until the next set of STEPS
+    else if ((pseudo_axis->skip_counter <= (2 * pseudo_axis->steps_for_ratio)) || (pseudo_axis->changes_to_skip == 0))
+    {   
+        bool motor_pin_status = digitalRead(pseudo_axis->encoder.motor_pulse_pin);
+        // Serial.print("motor pin:    "); Serial.println(motor_pin_status);
+        digitalWrite((pseudo_axis->encoder).channel_a_out, motor_pin_status ^ HIGH);
```

Comment:
Based on this, the encoder output is the inversion of the motor pulse. In the actual firmware, we increment our encoder count on the `RISING` edge of the encoder. Here you're returning true (and then incrementing the encoder count) on the rising edge of the motor therefore the `FALLING` edge of the encoder output.
There's nothing necessarily wrong with that (it just means the pseudo gantry will update it's encoder count a pulse width earlier than the actual firmware). Just wanna make sure that was intentional and you were aware of it.

Reply:
Actually, that was not my intention, so i will change it to: `digitalWrite((pseudo_axis->encoder).channel_a_pin, motor_pin_status);`